src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareUtils.cs

```diff
@@ -24,15 +25,24 @@ internal static class SchemaCompareUtils
     {
         internal static DacDeployOptions CreateSchemaCompareOptions(DeploymentOptions deploymentOptions)
         {
-            System.Reflection.PropertyInfo[] deploymentOptionsProperties = deploymentOptions.GetType().GetProperties();
+            PropertyInfo[] deploymentOptionsProperties = deploymentOptions.GetType().GetProperties();
 
             DacDeployOptions dacOptions = new DacDeployOptions();
             foreach (var deployOptionsProp in deploymentOptionsProperties)
             {
                 var prop = dacOptions.GetType().GetProperty(deployOptionsProp.Name);
                 if (prop != null)
                 {
-                    prop.SetValue(dacOptions, deployOptionsProp.GetValue(deploymentOptions));
+                    var val = deployOptionsProp.GetValue(deploymentOptions);
+                    var selectedVal = val.GetType().GetProperty("Value").GetValue(val);
+
+                    // The integer value coming from the DeploymentOptions is of Int64 which overflows to set to DacDeployOptions.
```

Comment:
What does this mean? You should be explaining why this conversion is necessary. And I'd also explain why it isn't a concern to be converting an int64 into an int32 - the convert function is going to throw if it's given a value that's too large 

`Run-time exception (line 10): Value was either too large or too small for an Int32.`

Reply:
Here the DacDeployOptions(DacFx model) has the int type of 32bin signed integer. when the DeploymentOptionProperty model coming back from ADS, these int values are as Int64 (really not sure what made this, any idea). when preparing the DacDeployOptions model with these return values, it is throwing an error **{"Object of type 'System.Int64' cannot be converted to type 'System.Int32'."}**. 

Since the default value is coming from DacFx as Int32 and is not going to change in ADS (as these fields are not available to edit in UI). I believe the exception won't be an issue here! Thanks

Reply:
I'm guessing it's because the JSON RPC layer just transfers numbers and not anything about them so it defaults to Int64 on the C# side. But in general leaving stuff up to a guess and hoping things work out isn't a great practice so I'd suggest : 

1. Spending a bit of time digging into it and finding out whether my guess above was correct or whether it's an issue elsewhere
2. Whatever it ends up being explain in the comment exactly why this needs to be done and why it's safe to assume this will work.

Reply:
https://stackoverflow.com/questions/8297541/how-do-i-change-the-default-type-for-numeric-deserialization

https://web.archive.org/web/20160606030722/http://json.codeplex.com/workitem/20832

With some time spent on researching the reason, it came that "JSON by default reads integer values as Int64 because there is no way to know whether the value should be Int32 or Int64, and Int64 is less likely to overflow. For a typed property the deserializer knows to convert the Int64 to a Int32" but here the type is object, so it gives as 64bit which needs a conversion. Thanks, will dosome more research if there is any other limitations and update comment more clear.